### DISCUSSION TITLE
SmartAutoClicker-1.5.3

### DISCUSSION THREAD

**Discussion by Nain57** [Nain57/Smart-AutoClicker#174] [Announcements]
## Bug fixes
* Fix a problem leading to scenario processing not waiting for an action to be completed. This should fix https://github.com/Nain57/Smart-AutoClicker/issues/168 
* Fix a possible crash on rotation
* Various stability improvements on the detection loop.

## Android 13
* The notification permission is now part of the permission dialog and it is optional. According to Google, denying this permission will not change the service lifecycle.

## New feature: Debug Report
In order to help you improve the performances of your scenario, you can now generate a debugging report of your scenario. If the option is enabled, once you are done with your detection session and press pause, the debug report dialog will be shown.
This report contains all timings for all detections, events executions; the number of hit/miss for each conditions and their confidence rate...

It can be enabled in the new debug menu, found in the top right corner of your event list. Please note that debugging will impact the performances a bit. So use it only when tuning your scenario !

**Comment by RaufSab** [comment]
Is the detection & activation speed amongst the stability improvements? Is the speed as 1.5.2 version or is it faster?

**Reply by Nain57** [reply]
No performance improvements, only stability fixes.

**Comment by RaufSab** [comment]
Do you know any similar autoclicker like yours for the iPhone?
Would you consider making this clicker for ios system?

**Reply by Nain57** [reply]
* 1: I don't know any.
* 2: No other OS than Android will be supported for this application.
